digitalocean/database/resource_database_opensearch_config_test.go

```diff
@@ -0,0 +1,48 @@
+package database_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/digitalocean/terraform-provider-digitalocean/digitalocean/acceptance"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccDigitalOceanDatabaseOpensearchConfig_Basic(t *testing.T) {
+	name := acceptance.RandomTestName()
+	dbConfig := fmt.Sprintf(testAccCheckDigitalOceanDatabaseClusterOpensearch, name, "3.7")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      testAccCheckDigitalOceanDatabaseClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(testAccCheckDigitalOceanDatabaseOpensearchConfigConfigBasic, dbConfig, true, 10, "1"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("digitalocean_database_opensearch_config.foobar", "ism_enabled", "true"),
+					resource.TestCheckResourceAttr("digitalocean_database_opensearch_config.foobar", "ism_history_max_age_hours", "10"),
+					resource.TestCheckResourceAttr("digitalocean_database_opensearch_config.foobar", "ism_history_max_docs", "1"),
+				),
+			},
+			{
+				Config: fmt.Sprintf(testAccCheckDigitalOceanDatabaseOpensearchConfigConfigBasic, dbConfig, false, 1, "9223372036854776000"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("digitalocean_database_opensearch_config.foobar", "ism_enabled", "false"),
+					resource.TestCheckResourceAttr("digitalocean_database_opensearch_config.foobar", "ism_history_max_age_hours", "1"),
+					resource.TestCheckResourceAttr("digitalocean_database_opensearch_config.foobar", "ism_history_max_docs", "9223372036854776000"),
```

Comment:
Seems like this is an issue with the documentation, but if I try to use this value the API rejects it:

```
│ Error: Error updating Opensearch configuration: PATCH https://api.digitalocean.com/v2/databases/d62c44d8-772b-4876-9bd9-25a03115277b/config: 422 (request "16de8945-d643-463a-a848-321b5c83c177") invalid 'user_config': invalid input for opensearch: 9223372036854776000 is greater than the maximum of 9223372036854775807
```

Reply:
Decided to change the type for this field in godo from uint64 to int64.
The max value will be updated to 9223372036854775807 in our API doc.

Reply:
@andrewsomething I updated the type here: [fa9a4ad](https://github.com/digitalocean/terraform-provider-digitalocean/pull/1234/commits/fa9a4ad88bef8c687dd2ff442205421c110aa2dc) :pray: 
Thanks a lot for catching this :+1: